test/test_datasets.py

```diff
@@ -1644,6 +1648,23 @@ def test_classes(self, config):
             assert len(dataset.classes) == len(info["classes"])
             assert all([a == b for a, b in zip(dataset.classes, info["classes"])])
 
+    def test_allow_empty(self):
```

Comment:
Not sure this is the intended way to test this given the test framework, but hopefully that's acceptable?
(I couldn't use `pytest.mark.parametrize()`)

Reply:
Just now realize what you probably meant: you couldn't parametrize over `allow_empty` and thus put both variants into this test. Although ugly, I don't think there is a better option right now.

Reply:
> I would remove the make_empty_class from the config and trigger on allow_empty

If I do this, I think I won't be able to check for the following, because the empty class wouldn't be created?

```py
        config["allow_empty"] = False
        with pytest.raises(FileNotFoundError, match="Found no valid file"):
            with self.create_dataset(config) as (dataset, info):
                pass
```

Reply:
(It is for similiar reasons that I didn't add `allow_empty` to the general config)

Reply:
Yes, fair enough. Disregard my comment.